File: sub-core/src/main/java/org/github/logger/p6spy/P6SpyLogger.java
package org.github.logger.p6spy;

import com.p6spy.engine.spy.appender.MessageFormattingStrategy;

import static org.apache.commons.lang3.StringUtils.isNotBlank;

public class P6SpyLogger implements MessageFormattingStrategy {
  @Override
  public String formatMessage(int connectionId, String now, long elapsed, String category, String prepared, String sql, String url) {
    return isNotBlank(sql) ? " Consume Time：" + elapsed + " ms " + now + "\n  Source URL：" + url + "\n Execute SQL：" + sql.replaceAll("[\\s]+", " ") + "\n" : "";
  }
}

File: sub-core/src/main/java/org/github/logger/P6SpyLogger.kt
package org.github.logger

import com.p6spy.engine.spy.appender.MessageFormattingStrategy

class P6SpyLogger: MessageFormattingStrategy {
  override fun formatMessage(connectionId: Int, now: String, elapsed: Long, category: String, prepared: String, sql: String, url: String) = sql.let {
    """Consume Time: ${"$elapsed".padStart(4, ' ')}ms | url: $url
      | Execute SQL: ${it.replace("\\s+".toRegex(), " ")}
    """.trimMargin()
  }
}
